jbmc/src/java_bytecode/java_trace_validation.cpp

```diff
@@ -181,6 +184,14 @@ static void check_rhs_assumptions(const exprt &rhs, const namespacet &ns)
     else if(constant_expr->get_value().empty())
       raise_error("RHS", *constant_expr);
   }
+  // check byte extract rhs structure
+  else if(rhs.id() == ID_byte_extract_little_endian)
```

Comment:
Why not try casting to byte_extract_exprt. Then you can use op() and offset() accessors?

Reply:
It's not implemented and I plan to implement it in a subsequent PR

Reply:
https://github.com/diffblue/cbmc/blob/develop/src/util/byte_operators.h#L28 ?

Reply:
Sorry, I meant can_cast_expr is not implemented for this class. I've a branch to do it so this can all be done a bit more elegantly. I don't want to make this PR larger and larger.